File: youlexuan_seckill_interface/src/main/java/com/offcn/seckill/service/SeckillGoodsService.java
package com.offcn.seckill.service;

import com.offcn.entity.PageResult;
import com.offcn.pojo.TbSeckillGoods;

import java.util.List;

/**
 * 秒杀商品服务层接口
 * @author Administrator
 *
 */
public interface SeckillGoodsService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbSeckillGoods> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);


	/**
	 * 增加
	*/
	public void add(TbSeckillGoods seckill_goods);


	/**
	 * 修改
	 */
	public void update(TbSeckillGoods seckill_goods);


	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbSeckillGoods findOne(Long id);


	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbSeckillGoods seckill_goods, int pageNum, int pageSize);


	//列出当前正在秒杀商品
	public List<TbSeckillGoods> findList();

	//从redis缓存，读取指定秒杀商品信息
	public TbSeckillGoods findOneFromredis(Long id);
}

File: youlexuan_seckill_web/src/main/webapp/js/service/seckillGoodsService.js
app.service('seckillGoodsService',function ($http) {
    //读取全部秒杀商品
    this.findList=function () {
      return  $http.get('/seckillGoods/findList.do');
    }

    //读取指定编号秒杀商品详情信息
    this.findOne=function (id) {
      return  $http.get('/seckillGoods/findOneFromRedis.do?id='+id);
    }
    
    //秒杀下单请求方法
    this.submitOrder=function (seckillId) {

      return  $http.get('/seckillOrder/submitOrder.do?seckillId='+seckillId);
    }
})